### DISCUSSION TITLE
Is it possible to prevent anon access to postgres function?

### DISCUSSION THREAD

**Discussion by Eldynn** [supabase/supabase#15921] [Questions] [answered]
I have a function in public to be able to call it via rpc in an edge function via the Supabase service role key.

But I don't want anon (or even authenticated) access to this function is it possible to have the same kind of granular control we have for table via RLS but for function?

**Comment by GaryAustin1** [comment, marked as answer]
You can check current_user in the code and use an if then.
You can modify the grants on your function to remove anon and authenticated.  
`revoke all privileges on function my_function from anon, authenticated;`
